### PR TITLE
TCVP-2075 Audit log entries (file history) ords support

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [TCVP-2075](https://justice.gov.bc.ca/jira/browse/TCVP-2075)
- Added mappers and functionality to insert and get audit log entries (file history) through the ORDS endpoints.
- Added new `AuditLogEntryType` enum.
- Added repository for saving and getting list of audit log entries through ORDS endpoints.
- Refactored file history controller to receive only `fileHistory` in the body of the request for inserting file history record.
- Refactored unit tests.
- Updated JJDispute to make occamDisputeId to be not nullable and updated .net code accordingly.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Ran oracle data api locally, inserted and retrieved audit log entries successfully using Swagger UI.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java
Comment (lfdgov) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java:17]: I'm a bit confused about what is going on here.  This would definitely be a breaking change, wouldn't it?  Can we have a new controller AuditLogEntryController, new methods? If there is a good reason to change the fileHistoryController thats ok, but otherwise we could just have AuditLogEntryController which is always pointing to Oracle, and fileHistoryController always H2.  Once we do the switch we are not going back to H2.

**Review by lfdgov — CHANGES_REQUESTED** [review]

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java
Comment (karahanb) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java:17]: Yes, this would be a breaking change if we merge this in as it is right now. However, I'm also working on the second part of this to refactoring .net apps that are affected by this change (TCVP-2077). That work is on a branch which is branched off of this one so once the second part of this work will be merged into this one (TCVP-2075), this will no longer a breaking change.

I don't see any value in adding extra code to trying to make it work with H2 at his point since our goal is completely switching to use ORDS. I created this PR without completing the fixing the breaking change part just because to make it easier to review.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/FileHistory.java
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/FileHistory.java:46]: By default, enums are serialized as the ordinal number (int). You may want to add `@Enumerated(EnumType.STRING)` to use the name of the enum as values in the database.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java
Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java:17]: Yeah, thanks for keeping the PRs shorter - it makes things a lot easier. I marked this PR as a Breaking Change so to remember not to merge it in until it's complete with your other work. When you make your other PRs, set them to merge into this branch instead of main so we can see the differences, then when all changes are ready, this branch can be merged all at once.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/FileHistory.java
Comment (karahanb) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/FileHistory.java:46]: This is true for storing the enum in H2 database as string but we don't store it in H2 anymore and only sending this to ORDS endpoint and get back from it. In this case, mapper takes care of converting it to the string value as auditLogEntryTypeCd. I haven't added the long descriptions for audit log entry enums for now. (some of them are really long) If we add that in the future, it can be useful to add toShortName() helper.

Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/FileHistory.java:46]: Ah yes, right you are.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java
Comment (karahanb) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/FileHistoryController.java:17]: Yes, this was the approach that I was thinking exactly. Will do that.

**Pull request merged**
